cartoframes/datasets.py

```diff
@@ -140,6 +141,20 @@ def _create_table_query(self, with_lonlat=None):
         create_query = '''CREATE TABLE {table_name} ({cols})'''.format(table_name=self.table_name, cols=cols)
         return create_query
 
+    def _recursive_read(self, query, retry_times=3):
+        try:
+            return self.cc.copy_client.copyto_stream(query)
+        except CartoRateLimitException as err:
+            if retry_times > 0:
+                retry_times -= 1
+                warn('Read call rate limited. Waiting {s} seconds'.format(s=err.retry_after))
+                time.sleep(err.retry_after)
```

Comment:
This is sweet. I didn't know this info was in the response header.

Reply:
:heart: 

We are returning:
- Carto-Rate-Limit-Limit: total allowed requests
- Carto-Rate-Limit-Remaining: remaining requests
- Retry-After: seconds until next available request (nothing if the current request is allowed)
- Carto-Rate-Limit-Reset: seconds until the limit will reset to its maximum capacity